### PR TITLE
Http api 1.0 rc changes

### LINKED ISSUE #48: For connection.js, in all methods use camelCase named parameters (and preferably no objects)

### PR DESCRIPTION
**Pull request by TimDaub**
- Changes the `Connection` model to the agreed on HTTP API breaking changes as outlines [here](https://github.com/bigchaindb/bigchaindb/blob/babe5311a46684ef7679ec843b1f600a7bceed38/docs/upgrade-guides/v0.10--%3Ev1.0.md)
- adds tests for all changes
- adds sinon.js for mocking
- changed path describing keys to camelCase
- added constant.js (similar to conftest.py)
- added integration tests that run against bdb
- updated ascribe-eslint package
- [x] test and fix asset endpoint
- [x] test and fix outputs endpoint

More info: 

- Fixes #48

### CODE REVIEW

File: test/connection/test_connection.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    Object.keys(endpoints).forEach((endpointName) => {
+        const url = conn.getApiUrls(endpointName)
+        const expected = API_PATH + endpoints[endpointName]
+        t.is(url, expected)
+    })
+})
+
+
+// TODO: Find out how to mock `request`
+test.skip('Request with custom headers', t => {
```
Comment (TimDaub) [inline test/connection/test_connection.js]: Import/export from parent module (index.js)

Comment (TimDaub) [inline test/connection/test_connection.js]: Fixed this

**Comment by TimDaub** [comment]
> Add integration test for asset endpoint :diamond:

:(

**Comment by TimDaub** [comment]
Please note that in [Connection.js I broke the interface to use camelCased parameters](https://github.com/bigchaindb/js-bigchaindb-driver/issues/48) for some methods already.

It will require me to fix the rest of those before releasing a 1.0 compatible release.

File: test/constants.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import test from 'ava'
+import { Transaction, Ed25519Keypair } from '../src'
+// TODO: Find out if ava has something like conftest, if so put this there.
+
+// NOTE: We cast `Math.random()` to a string, as sometimes Javascript simply
+// yields a slightly different float during runtime, lol
```
Comment (vrde) [inline test/constants.js]: What do you mean by "slightly different float"?

**Comment by TimDaub** [comment]
> Apart from the first test in test/connection/test_connection.js, I'm not sure I understand how the rest of the tests in the same file work.

They check if for given input parameters, the tested function calls the request method correctly.
Example:

```
Call getBlocks(transactionId='abc', status='VALID`)
return {
  query: {
    transaction_id: 'abc',
    status: 'VALID'
  }
}
```


> I think hard-coding Alice's pubkey (and outputs) is not a good idea since tests run in parallel. In this case we can have problems processing the unspents—e.g. I want to create a test where I transfer something from Alice to Bob. If Bob's pubkey is constant, tests can influence each others and assertions like:
> 
> unspentst-1(Bob) = unspentst(Bob) + 1
> can easily fail.

OK, I can fix this.

**Comment by TimDaub** [comment]
> remove extra parenthesis :nail_care:

https://github.com/ascribe/javascript/issues/56

File: test/integration/test_integration.js [hunk 3]
```diff
@@ -112,137 +112,130 @@ test('Valid TRANSFER transaction with multiple Ed25519 inputs', t => {
...
+    const trentCondition = Transaction.makeEd25519Condition(trent.publicKey)
+    const trentOutput = Transaction.makeOutput(trentCondition)
+
 
     const createTx = Transaction.makeCreateTransaction(
         asset(),
         metaData,
-        [aliceOutput, aliceOutput],
-        alice.publicKey
+        [carolOutput, carolOutput],
```
Comment (vrde) [inline test/integration/test_integration.js]: who is Carol?

**Comment by libscott** [comment]
Why is it connection/index.js rather than just connection.js?

**Comment by TimDaub** [comment]
> Why is it connection/index.js rather than just connection.js?

https://github.com/bigchaindb/js-bigchaindb-driver/issues/57

**Pull request merged**